src/lib/price.ts

```diff
@@ -4,10 +4,17 @@ import { type Chain, toDefiLlamaChain } from '@lib/chains'
 import { mulPrice, sum } from '@lib/math'
 import type { Token } from '@lib/token'
 import { isNotNullish } from '@lib/type'
+import { formatUnits } from 'viem'
+
+// Defillama prices API requires a prefix to know where the token comes from.
+// It can be a chain or a market provider like coingecko
+export function getTokenKey(token: { chain: Chain; address: string }): string {
+  if (!token.chain || !token.address) {
+    console.error('[getTokenKey]: Invalid token', token)
+    throw new Error('Invalid token')
+  }
 
-// Defillama prices API requires a prefix to know where the token comes from
-export function getTokenKey(token: { chain: Chain; address: string }) {
-  return `${toDefiLlamaChain[token.chain] || token.chain}:${token.address.toLowerCase()}`
```

Comment:
```ts
return `${toDefiLlamaChain[token.chain] || token.chain}:${token.address.toLowerCase()}`
```

handles more cases

Reply:
sweet, I'll switch to that. 

Is the space safe here? https://github.com/llamafolio/llamafolio-api/blob/fa137913131cda6c85d427c63dacc62d8c407e28/src/lib/chains.ts#L320